src/vmshepherd_aws_drivers/aws_iaas_driver.py

```diff
@@ -0,0 +1,101 @@
+from itertools import chain
+import aiobotocore
+from botocore.exceptions import ClientError
+from typing import Any
+from vmshepherd.iaas import AbstractIaasDriver, Vm, VmState
+
+
+class AwsIaaSDriver(AbstractIaasDriver):
+
+    def __init__(self, config):
+        self.config = config
+
+    async def list_vms(self, preset_name):
+        '''
+        List VMs by preset name
+        :arg present_name: string
+        '''
+        session = aiobotocore.get_session()
+        async with session.create_client('ec2') as client:
+            paginator = client.get_paginator('describe_instances')
+            filters = [
+                {
+                    'Name': 'tag:aws:autoscaling:groupName',
+                    'Values': [preset_name]
+                }
+            ]
+            instances = []
+            async for result in paginator.paginate(
+                PaginationConfig={'PageSize': self.config.get('ec2_page_size', 5)},
+                Filters=filters
+            ):
+                instances = chain(instances, *[vms['Instances'] for vms in result['Reservations']])
+
+        return [self._map_vm_structure(instance) for instance in instances]
+
+    async def create_vm(self):
+        '''
+        NotImplemented - preset is managed by ASG
+        '''
+        pass
+
+    async def get_vm(self, vm_id: Any):
```

Comment:
I don't see the point of such selective use of type annotations, especially when they states `Any`. I would greatly appreciate type annotations all these interface methods.